inf_post/src/checks.cpp

```diff
@@ -0,0 +1,84 @@
+#include "checks.h"
+#include <stack>
+
+bool is_valid(std::string& s)
+{
+    return (!(!is_right_string(s) || !is_balanced(s) 
+            || !operand_operator(s) || is_operator(s[0])));
+}
+
+bool operand_operator(std::string& s)
+{
+    int operand_count = 0, operator_count = 0;
+    for(int i = 0; i < s.size(); ++i)
+    {
+        if(is_operand(s[i]))
+	{
+	    operand_count++;
+	}
+	if(is_operator(s[i]))
+	{
+	    operator_count++;
+	}
+    }
+    return (operand_count-operator_count == 1);
+}
+
+bool is_right_string(std::string& s)
+{
+    for(int i = 0; i< s.size(); ++i)
+    {
+        if(!(is_operand(s[i])) && !(is_operator(s[i])) 
+		        && s[i] != '(' && s[i] != ')')
+	{
+            return false;
+	}
+    }
+    return true;
+}
+
+bool is_infix(std::string& s)
+{
+    return !is_operator(s[s.size()-1]);
```

Comment:
string has '[back](https://en.cppreference.com/w/cpp/string/basic_string/back)' which allows to access the last element